vm_method.c

```diff
@@ -1902,13 +1912,17 @@ set_method_visibility(VALUE self, int argc, const VALUE *argv, rb_method_visibil
 	return;
     }
 
-    for (i = 0; i < argc; i++) {
-	VALUE v = argv[i];
-	ID id = rb_check_id(&v);
-	if (!id) {
-	    rb_print_undef_str(self, v);
+    if (argc == 1 && RB_TYPE_P(argv[0], T_ARRAY)) {
```

Comment:
Please use `rb_check_array_type` that will do implicit conversion with `to_ary` if it can be applied